demos/speech_recognition_demo/python/ctcdecode-numpy/ctcdecode_numpy/yoklm/memory_section.cpp

```diff
@@ -85,15 +94,23 @@ MemorySection MemorySection::get_and_drop_prefix(size_t size) {
   const uint8_t * ptr_orig = ptr_;
   size_ -= size;
   ptr_ += size;
-  return MemorySection(ptr_orig, size, owner_);
+  return MemorySection(ptr_orig, base_ptr_, size, owner_);
 }
 
 void MemorySection::reset() {
   size_ = 0;
-  ptr_ = nullptr;
+  ptr_ = base_ptr_ = nullptr;
   owner_.reset();
 }
 
+void MemorySectionBitArray::set_stride(int stride) {
+  stride_ = stride;
+  // Dividing max(size_t) by 2 to be sure there's no overflows.
+  if (stride != 0 && (stride <= 0 || size() / size_t(stride) >= std::numeric_limits<size_t>::max() / 8 / 2))
+    throw std::range_error("Array size exceeds size_t in MemorySectionBitArray. Broken LM file?");
```

Comment:
1) The cause seems to be that SWIG Python adapter doesn't catch C++ exceptions by default. I thought it does. So the result is probably compiler-dependent (as Python makes a C call, not C++). I'll add exception handler in a commit to a different PR to avoid merge conflict, and because this SWIG adapter is outside yoklm.

2) I think it isn't good to litter to stderr: an app might not want anything on stderr if it catches exceptions itself. The proper way is to fix SWIG adapter, as I said.

Reply:
Added a commit to this PR: https://github.com/openvinotoolkit/open_model_zoo/pull/2108 .